src/Orleans.GrainDirectory.Redis/RedisGrainDirectory.cs

```diff
@@ -89,11 +89,24 @@ public async Task<GrainAddress> Register(GrainAddress address)
 
         public async Task Unregister(GrainAddress address)
         {
-            var value = JsonConvert.SerializeObject(address);
+            var value = JsonSerializer.Serialize(address);
 
             try
             {
-                var result = (int) await this.database.ScriptEvaluateAsync(this.deleteScript, new { key = GetKey(address.GrainId), val = JsonConvert.SerializeObject(address) });
+                if (address.MembershipVersion == MembershipVersion.MinValue)
```

Comment:
There should be a comment here explaining that this indicates that the provided `address` was not returned from the directory and must have been constructed externally - or some kind of comment.

Generally, we should aim for more descriptive comments.